cloud/amazon/ec2_vpc_vgw_facts.py

```diff
@@ -0,0 +1,162 @@
+    description: The virtual gateways for the account
+    returned: always
+    type: list
+
+changed:
+    description: True if listing the virtual gateways succeeds
+    type: bool
+    returned: always
+'''
+
+try:
+    import json
+    import botocore
+    import boto3   
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_virtual_gateway_info(virtual_gateway):
+    virtual_gateway_info = {'VpnGatewayId': virtual_gateway['VpnGatewayId'],
+                             'State': virtual_gateway['State'],
+                             'Type': virtual_gateway['Type'],
+                             'VpcAttachments': virtual_gateway['VpcAttachments'],
+                             'Tags': virtual_gateway['Tags']                                                              
+               }
+    return virtual_gateway_info
+
+
+def list_virtual_gateways(client, module):
+    dryrun = module.params.get("DryRun")
+    all_virtual_gateways_array = []
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filters'] = []
+        for key, value in module.params.get('filters').iteritems():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filters'].append(temp_dict)
+
+    if module.params.get("DryRun"):
+        params['DryRun'] = module.params.get("DryRun")
+
+    if module.params.get("VpnGatewayIds"):
+        params['VpnGatewayIds'] = module.params.get("VpnGatewayIds")
+
+    try:
+        all_virtual_gateways = client.describe_vpn_gateways(**params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    for virtual_gateway in all_virtual_gateways['VpnGateways']:
+        all_virtual_gateways_array.append(get_virtual_gateway_info(virtual_gateway))
+
+    module.exit_json(virtual_gateways=all_virtual_gateways_array)
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            filters = dict(type='dict', default=None, ),
+            DryRun = dict(type='bool', default=False),
+            VpnGatewayIds = dict(type='list', default=None)
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+     # Validate Requirements
+    if not HAS_BOTO3:
+        module.fail_json(msg='json and botocore/boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        connection = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.NoCredentialsError, e:
+        module.fail_json(msg="Can't authorize connection - "+str(e))
+
+    # call your function here
+    results = list_virtual_gateways(connection, module)
+    
+    module.exit_json(result=results)
```

Comment:
You need to change the boto result to be ansible friendly by changing from CamelCase to snake_case.

Use the helper function 'camel_dict_to_snake_dict'.  See GUIDELINES and ec2_snapshot_facts for example